utils/io/httputils/pollingexecutor.go

```diff
@@ -0,0 +1,60 @@
+package httputils
+
+import (
+	"time"
+
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+type PollingAction func() (shouldStop bool, responseBody []byte, err error)
+
+type PollingExecutor struct {
```

Comment:
Do we really need ithe PollingExecutor Looks like we already have the RetryExecutor. Both are generic.
Can we use the RetryExecutor instead? If not, can we merge the two utils?
If we do decide to keep the PollingExecutor, please let me know and I'll finish reviewing it, because I'm pausing the review for this